### ISSUE TITLE
Error: Could Not Rename: ... to ...

### ISSUE THREAD

**Issue by Ar7eniyan** [tldr-pages/tldr-c-client#95]
## Reproduction Rate
On each `tldr <command>` run without local database

## Steps to Reproduce
1. `rm -rf ~/.tldrc`
2. `tldr tldr`

## Result
Couple seconds of delay (downloading the tldr database), then the error message `Error: Could Not Rename: /tmp/tldrXXXXXX/tldr-main to /home/<user>/.tldrc/tldr/`, then the actual tldr page.
`strace` shows that the [rename](https://github.com/tldr-pages/tldr-c-client/blob/903b38d5fa2b97ee8b72e151efbe6b797618059d/src/local.c#L213) call fails with EXDEV (Invalid cross-device link).

## Expected Result
Normal output witout error

## Additional Information
```
$ tldr --version
tldr v1.5.0 (v1.5.0-3-g903b38d)
Copyright (C) 2016 Arvid Gerstmann
Source available at https://github.com/tldr-pages/tldr-c-client
```
Arch Linux, tldr installed from [tldr-git](https://aur.archlinux.org/packages/tldr-git)

**Comment by MasterOdin** [comment]
To help me understand the actual bug and fixing it, for your setup, do you have `/tmp` and `/home` mounted to different drives?

**Comment by Ar7eniyan** [comment]
@MasterOdin Yes, `/tmp` is `tmpfs` and `/home` is `ext4`

**Comment by eliminmax** [comment]
Ran into this issue myself, and found [another project](https://github.com/andrewkroh/gvm/issues/1) with the same issue. Apparently, it's a limitation of the `rename()` library call, which can be worked around by copying the source to the destination, then deleting the source once it succeeds. This is what the `mv` command does.

My experience with C is very limited, and given that it does not seem like there's a simple, standard, cross-platform way to copy a file, I don't think I am the right person to fix this.

**Comment by Ar7eniyan** [comment]
After some research, I haven't found any reliable cross-platform way to move a directory between different FS, doing it manually in C is error-prone and complicated. Seems like the best way to deal with it is to download/unpack zip file to `$HOME/.tldrc/tldr/tempXXXXXX`. Or if it really needs to be in `/tmp` (which it doesn't), directory moving can be done using `system("/bin/mv ...")`

**Comment by dylanmtaylor** [comment]
I'm also hitting this. 

Error: Could Not Rename: /tmp/tldrFB8aMw/tldr-main to /home/taylord/.tldrc/tldr/

My /home is on a separate partition as well.
